jaxrs/src/test/java/io/airlift/jaxrs/TestRequestScopedBinding.java

```diff
@@ -0,0 +1,141 @@
+{
+    @Test
+    public void testRequestScopedInjection()
+    {
+        Injector injector = startServer();
+        try {
+            HttpClient httpClient = injector.getInstance(Key.get(HttpClient.class, ForTest.class));
+            URI serverUri = injector.getInstance(HttpServerInfo.class).getHttpUri();
+
+            assertEquals(executeRequest(httpClient, serverUri, "Bob"), "Bob@127.0.0.1");
+            assertEquals(executeRequest(httpClient, serverUri, "Alice"), "Alice@127.0.0.1");
+        }
+        finally {
+            injector.getInstance(LifeCycleManager.class).stop();
+        }
+    }
+
+    private String executeRequest(HttpClient httpClient, URI baseUri, String name)
+    {
+        Request request = prepareGet()
+                .setUri(baseUri.resolve("/hello"))
+                .setHeader("X-Name", name)
+                .build();
+        return httpClient.execute(request, createStringResponseHandler()).getBody();
+    }
+
+    @Path("/hello")
+    public static class MyResource
+    {
+        @GET
+        public String returnUserScopedInfo(@Context RemoteAddr remoteAddr, @Context HeaderValue headerValue)
+        {
+            return headerValue.getValue() + "@" + remoteAddr.value();
+        }
+    }
+
+    @Retention(RUNTIME)
+    @BindingAnnotation
+    public @interface ForTest {}
+
+    public record RemoteAddr(String value) {}
+
+    public static class HeaderValue
+    {
+        private final String value;
+
+        public HeaderValue(String value)
+        {
+            this.value = requireNonNull(value, "value is null");
+        }
+
+        public String getValue()
+        {
+            return value;
+        }
+    }
+
+    private Injector startServer()
+    {
+        return new Bootstrap(
+                binder -> {
+                    jaxrsBinder(binder).bind(MyResource.class);
+                    httpClientBinder(binder).bindHttpClient("test", ForTest.class);
+                    jaxrsBinder(binder).bindRequestScoped(RemoteAddr.class, RemoteAddrFactory.class);
+                    jaxrsBinder(binder).bindRequestScoped(HeaderValue.class, HeaderValueFactory.class);
+                },
+                new TestingNodeModule(),
+                new TestingHttpServerModule(),
+                new JaxrsModule(),
+                new JsonModule())
+                .quiet()
+                .initialize();
+    }
+
+    private record RemoteAddrFactory(HttpServletRequest request)
+            implements Supplier<RemoteAddr>
+    {
+        private RemoteAddrFactory(@Context HttpServletRequest request)
+        {
+            this.request = requireNonNull(request, "request is null");
+        }
+
+        @Override
+        public RemoteAddr get()
+        {
+            return new RemoteAddr(request.getRemoteAddr());
+        }
+    }
+
+    private static class HeaderValueFactory
+            implements Supplier<HeaderValue>
+    {
+        private final HttpHeaders headers;
+
+        @Inject
```

Comment:
This shouldn't be needed as this object is not constructed with Guice

Reply:
Right, dropped.